src/renderer/shared/api/governance/on-chain/service/claimScheduleService.ts

```diff
@@ -0,0 +1,462 @@
+  referendums: Record<ReferendumId, ReferendumInfo>,
+  trackId: TrackId,
+  tracks: Record<TrackId, TrackInfo>,
+  voting: CastingVoting,
+  voteLockingPeriod: BlockHeight,
+  undecidingTimeout: BlockHeight,
+): ClaimableLock[] {
+  const priorLock: ClaimableLock = {
+    claimAt: { type: 'at', block: voting.casting.prior.unlockAt } as ClaimTime,
+    amount: voting.casting.prior.amount,
+    affected: [{ trackId, type: 'track' } as AffectTrack],
+  };
+
+  const standardVotes = Object.entries(voting.casting.votes) as [string, StandardVote][];
+
+  const standardVoteLocks = standardVotes.map(([referendumId, standardVote]) => {
+    const estimatedEnd = maxConvictionEndOf(
+      currentBlockNumber,
+      trackId,
+      tracks,
+      standardVote,
+      voteLockingPeriod,
+      undecidingTimeout,
+      referendums[referendumId],
+    );
+
+    return {
+      // we estimate whether prior will affect the vote when performing `removeVote`
+      claimAt: {
+        type: 'at',
+        block: Math.max(estimatedEnd, (priorLock.claimAt as ClaimTimeAt).block),
+      } as ClaimTime,
+      amount: standardVote.balance,
+      affected: [{ trackId, type: 'vote', referendumId } as AffectVote],
+    } as ClaimableLock;
+  });
+
+  return priorLock.amount.isZero() ? standardVoteLocks : [priorLock, ...standardVoteLocks];
+}
+
+function delegatingClaimableLocks(trackId: TrackId, voting: DelegatingVoting): ClaimableLock[] {
+  const delegationLock = {
+    claimAt: { type: 'until' } as ClaimTimeUntil,
+    amount: voting.delegating.balance,
+    affected: [],
+  } as ClaimableLock;
+
+  const priorLock = {
+    claimAt: { type: 'at', block: voting.delegating.prior.unlockAt } as ClaimTimeAt,
+    amount: voting.delegating.prior.amount,
+    affected: [{ type: 'track', trackId } as AffectTrack],
+  } as ClaimableLock;
+
+  return priorLock.amount.isZero() ? [delegationLock] : [delegationLock, priorLock];
+}
+
+function maxConvictionEndOf(
+  currentBlockNumber: BlockHeight,
+  trackId: TrackId,
+  tracks: Record<TrackId, TrackInfo>,
+  vote: AccountVote,
+  voteLockingPeriod: BlockHeight,
+  undecidingTimeout: BlockHeight,
+  referendum?: ReferendumInfo,
+): BlockHeight {
+  // referendum is not in the map, which means it is cancelled and votes can be unlocked immediately
+  return referendum
+    ? referendumMaxConvictionEnd(referendum, trackId, tracks, vote, voteLockingPeriod, undecidingTimeout)
+    : currentBlockNumber;
+}
+
+function referendumMaxConvictionEnd(
+  referendum: ReferendumInfo,
+  trackId: TrackId,
+  tracks: Record<TrackId, TrackInfo>,
+  vote: AccountVote,
+  voteLockingPeriod: BlockHeight,
+  undecidingTimeout: BlockHeight,
+): BlockHeight {
+  if (onChainUtils.isOngoing(referendum)) {
+    return maxOngoingConvictionEnd(referendum, vote, trackId, tracks, voteLockingPeriod, undecidingTimeout);
+  }
+  if (onChainUtils.isRejected(referendum)) {
+    return maxCompletedConvictionEnd(vote, 'nay', referendum.since, voteLockingPeriod);
```

Comment:
Maybe use for `nay` and `aye`?

Reply:
Initially I made it this way, but found many `const nums` for such simple values and it's only used in this service